source/MRMesh/MRContoursCut.cpp

```diff
@@ -2190,6 +2191,29 @@ CutMeshResult cutMesh( Mesh& mesh, const OneMeshContours& contours, const CutMes
     return res;
 }
 
+Expected<FaceBitSet> cutMeshByContour( Mesh& mesh, const Contour3f& contour, const AffineXf3f& xf )
+{
+    std::vector<MeshTriPoint> surfaceLine;
+    for ( const auto& pt : contour )
+    {
+        PointOnFace projPt;
+        if ( !mesh.projectPoint( xf( pt ), projPt ) )
+        {
+            return unexpected( "Cannot project point to mesh" );
+        }
+        surfaceLine.push_back( mesh.toTriPoint( projPt ) );
+    }
+
+    auto meshContour = convertMeshTriPointsToMeshContour( mesh, surfaceLine );
+    if ( !meshContour )
+        return unexpected( "Cannot convert tri points to mesh contour: " + meshContour.error() );
+
+    auto cutRes = cutMesh( mesh, { *meshContour } );
```

Comment:
cut mesh can also be invalid, pls check `bool valid = cutRes.fbsWithCountourIntersections.none()`